plugins/es-repository-s3/src/test/java/org/elasticsearch/repositories/s3/S3ClientSettingsTests.java

```diff
@@ -118,4 +118,10 @@ public void testCredentialsTypeWithAccessKeyAndSecretKeyAndSessionToken() {
         assertThat(credentials.getAWSSecretKey()).isEqualTo("secret_key");
         assertThat(credentials.getSessionToken()).isEqualTo("session_token");
     }
+
+    @Test
+    public void test_when_key_pair_is_not_provided_credentials_can_be_null() {
+        final S3ClientSettings settings = S3ClientSettings.getClientSettings(Settings.EMPTY);
+        assertThat(settings.credentials).isNull();
+    }
```

Comment:
Shouldn't we rather verify hashCode don't throw NPE by asserting 2 empty settings are equal?
The Nullable annotation should be already enough to highlight that the credentials can be null.

Reply:
Thank you, updated